imagepipeline/src/main/java/com/facebook/imagepipeline/producers/HttpUrlConnectionNetworkFetcher.java

```diff
@@ -47,37 +54,33 @@ public void fetch(final FetchState fetchState, final Callback callback) {
         new Runnable() {
           @Override
           public void run() {
-            HttpURLConnection connection = null;
-            Uri uri = fetchState.getUri();
-            String scheme = uri.getScheme();
-            String uriString = fetchState.getUri().toString();
-            while (true) {
-              String nextUriString;
-              String nextScheme;
+              HttpURLConnection connection = null;
+              Uri uri = fetchState.getUri();
+              String uriString = uri.toString();
               InputStream is;
               try {
-                URL url = new URL(uriString);
-                connection = (HttpURLConnection) url.openConnection();
-                nextUriString = connection.getHeaderField("Location");
-                nextScheme = (nextUriString == null) ? null : Uri.parse(nextUriString).getScheme();
-                if (nextUriString == null || nextScheme.equals(scheme)) {
+                  connection = createGetConnection(uriString);
+                  int redirectCount = 0;
+                  while (connection.getResponseCode() / 100 == 3 && redirectCount < MAX_REDIRECT_COUNT) {
+                      uriString = connection.getHeaderField("Location");
+                      connection.disconnect();
+                      connection = createGetConnection(uriString);
+                      ++redirectCount;
+                  }
+                  if (redirectCount > MAX_REDIRECT_COUNT) {
```

Comment:
How can `redirectCount` be larger than `MAX_REDIRECT_COUNT`? When `redirectCount` is equal to `MAX_REDIRECT_COUNT`, the statements inside `while` block will not be executed so its value will not increase.